src/renderer/features/wallets/DerivationsAddressModal/DerivationsAddressModal.tsx

```diff
@@ -0,0 +1,124 @@
+import keyBy from 'lodash/keyBy';
+import { Dictionary } from 'lodash';
+
+import { BaseModal, Button, InfoLink, SmallTitleText } from '@renderer/shared/ui';
+import QrDerivationsGenerator from '@renderer/components/common/QrCode/QrGenerator/QrDerivationsGenerator';
+import { useToggle } from '@renderer/shared/lib/hooks';
+import { BaseAccount, ChainAccount, SigningType, WalletType } from '@renderer/shared/core';
+import { ShardAccount, ShardedAccount } from '@renderer/shared/core/types/account';
+import { useI18n } from '@renderer/app/providers';
+import { TROUBLESHOOTING_URL } from '@renderer/components/common/QrCode/common/constants';
+import { DdKeyQrReader } from '@renderer/pages/Onboarding/Vault/DdKeyQrReader/DdKeyQrReader';
+import { toAccountId, toAddress } from '@renderer/shared/lib/utils';
+import { walletModel } from '@renderer/entities/wallet';
+import { DdAddressInfoDecoded, DynamicDerivationRequestInfo } from '@renderer/components/common/QrCode/common/types';
+
+export type ShardedAccountWithShards = ShardedAccount & { shards: ShardAccount[] };
+type DerivationsAccounts = Omit<ShardedAccountWithShards | ChainAccount, 'accountId' | 'walletId' | 'id'>;
+
+const createDerivations = (accounts: DerivationsAccounts[]): DynamicDerivationRequestInfo[] => {
+  return accounts.reduce<DynamicDerivationRequestInfo[]>((acc, account) => {
+    if ('shards' in account) {
```

Comment:
Maybe it'd be better to create a function like `isAccountWithShards`  in 
 `account-utils` to check this condition everywhere?

Reply:
moved to accountUtils